syntax = "proto3";

package user.v1;

service UserService {
  rpc GetUser (GetUserRequest) returns (UserResponse);
  rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
}

message GetUserRequest {
  int32 id = 1;
}

message CreateUserRequest {
  int32 id = 1;
  string firstname = 2;
  string lastname = 3;
  string email = 4;
}

message UserResponse {
  int32 id = 1;
  string firstname = 2;
  string lastname = 3;
  string email = 4;
  string created_at = 5;
}

message CreateUserResponse {
  bool success = 1;
}